### DISCUSSION TITLE
UI Plugin - Civitai LORA Keyword Import/Storage

### DISCUSSION THREAD

**Discussion by StrayerJ** [lllyasviel/Fooocus#3482] [Ideas]
One thing that I always found very useful in the now abandoned EasyDiffusion, was the ability to automatically import a LORA's trigger words from Civitai.  If you rack up many LORA's, its impossible to keep track of the keywords.  I think something similar would be useful in Fooocus.  

In that same light, it would be also useful to have an easy "plugins" functionality for people to build their own UI elements on top of Fooocus, and share with each other, without a million forks out there.

I believe this is the code for the LORA Keyword Import:
https://github.com/easydiffusion/easydiffusion/blob/716f30fecb3497644eb1033cc63f965a19c3f871/ui/plugins/ui/model-tools.plugin.js#L10

![image](https://github.com/user-attachments/assets/fcd895f9-88f8-4e24-86cc-4e60775be711)

**Comment by mashb1t** [comment]
To be honest, I think it's too late for that. Fooocus by design didn't have an extension system and now that the frequency of PRs is dwindling i doubt that the effort would justify the invest needed to implement this.
